### ISSUE TITLE
`hab pkg hash` doesn't work with filenames with commas

### ISSUE THREAD

**Issue by miketheman** [habitat-sh/habitat#1151]
Discovered in `glibc`, related comments [here](https://github.com/habitat-sh/core-plans/pull/114#issuecomment-239665616).

It appears that the filename containing a comma is being strpped, and thereby ignored during hashing.

This ends up leading a malformed line in the package's `FILES`, like so:
```
...
591703eb337d2cd4ec2b6b7e718cd74f402b20fbfb9aa1e77fdab4d5a551e0ed  /hab/pkgs/miketheman/glibc/2.22/20160814125224/share/i18n/charmaps/ISO_6937.gz
  /hab/pkgs/miketheman/glibc/2.22/20160814125224/share/i18n/charmaps/ISO_8859-1,GL.gz
b0ef98622602f46452aaad3f2474f88126fdf06a1bdaa8077e3b763fd4672f87  /hab/pkgs/miketheman/glibc/2.22/20160814125224/share/i18n/charmaps/ISO_8859-SUPP.gz
...
```
The file appears to be included correctly in the package, but there's no hash, and a non-failing error message is emitted at build time, which is a little concerning.

---
- [x] The OS (including version) where you are running any of the Habitat commands.

```
# uname -a
Linux 194337aa6263 4.4.16-moby #1 SMP Tue Aug 9 17:20:17 UTC 2016 x86_64 GNU/Linux
```

- [x] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace:

		[default:/hab/cache/src/glibc-2.22:1]# RUST_LOG=debug RUST_BACKTRACE=1 hab pkg hash "./localedata/charmaps/ISO_8859-1,GL"
                DEBUG:hab: clap cli args: ["hab", "pkg", "hash", "./localedata/charmaps/ISO_8859-1,GL"]
                DEBUG:hab: remaining cli args: []
                error: File: './localedata/charmaps/ISO_8859-1' cannot be found
		

```
# hab --version
hab 0.8.0/20160708161531
```

**Comment by metadave** [comment]
Interesting. I was hoping this was already fixed via https://github.com/habitat-sh/habitat/pull/1076, but it appears to be related to how the [Clap](https://github.com/kbknapp/clap-rs) argument parser is handling input.

The `hab pkg hash` subcommand is defined [here](https://github.com/habitat-sh/habitat/blob/51d34435257f27553f6489d5a00f894f57bd1cb3/components/hab/src/cli.rs#L160-L164), which uses the `file_exists` function as an [input](http://kbknapp.github.io/clap-rs/clap/struct.Arg.html#method.validator) [validator](https://github.com/kbknapp/clap-rs/blob/master/examples/15_custom_validator.rs). I tested this morning against master via:

```
/src/target/debug/hab pkg hash "/src/foo/localedata/charmaps/ISO_8859-1,GL"
```

I really want to quoted path to work :-( The [file_exists](https://github.com/habitat-sh/habitat/blob/51d34435257f27553f6489d5a00f894f57bd1cb3/components/hab/src/cli.rs#L353-L359) function receives the input:

```
/src/foo/localedata/charmaps/ISO_8859-1
```

instead of the full filename, including the `,GL` "suffix"

```
/src/foo/localedata/charmaps/ISO_8859-1,GL
```

(The `/src/foo` prefix is from my local testing)

cc @habitat-sh/habitat-core-maintainers

**Comment by metadave** [comment]
I thought `use_delimiter` would help resolve the issue, but it didn't:

```
diff --git a/components/hab/src/cli.rs b/components/hab/src/cli.rs
index 0b41efa..c584e65 100644
--- a/components/hab/src/cli.rs
+++ b/components/hab/src/cli.rs
@@ -160,7 +160,7 @@ pub fn get() -> App<'static, 'static> {
             (@subcommand hash =>
                 (about: "Generates a blake2b hashsum from a target at any given filepath")
                 (aliases: &["ha", "has"])
-                (@arg SOURCE: +required {file_exists} "A filepath of the target")
+                (@arg SOURCE: +use_delimiter[false] +required {file_exists} "A filepath of the target")
             )
             (subcommand: sub_pkg_install().aliases(
                 &["i", "in", "ins", "inst", "insta", "instal"]))
```

**Comment by juliandunn** [comment]
Closed by #1250
